gnosis/safe-react
# NOTE: REMOVE PROD CHECK BEFORE MERGE - Rename xDai to Gnosis Chain

## What it solves
Resolves #3200

## How this PR fixes it
A redirection has been added for all `xdai:` prefixed URLs. Addressing each of the points in the issue as follows:

> Change the name to Gnosis Chain in config-service
> Change the shortName from xdai to gno in config-service
> Public RPC is the same, so no change to the wallet config

The chain config of xDai was copied over to the staging Django, renaming the name to 'Gnosis Chain' and `shortName` to 'gno'.

> Add a redirect from old xdai: URLs to gno:

When the URL is *not* a legacy link, any instances of `/app/xdai:` in the URL are replaced with `/app/gno:`.

> Check for any other places where we could be using the shortName xdai in the code

The deployment script was updated to reflect the new name and PR link `{{PR}}/gnosis-chain`.

> Local storage can keep using XDAI for prefixes to avoid migrating data

This code already existed. The local `CHAIN_ID` enum, used purely for edge cases, `XDAI` key was renamed to `GNOSIS_CHAIN`.

> Check that hardware wallets still work (networkName in the onboard config)

This will need to be tested by QA as I don't currently have access to a HW wallet.

> Adjust the legacy redirect from the xdai subdomain to go straight to gno shortName URLs

This will need to be addressed by DevOps. The `shortName` redirection above should cover this for the time being.

## How to test it

**PLEASE NOTE** that there is no official staging version of Gnosis Chain and the Safes you are viewing are taken from PRODUCTION xDai

- Open an `xdai:` prefixed URL and observe that it redirects to `gno:`. All functionality should remain the same, without bugs.
- `localStorage` should **not** generate a new key relating to Gnosis Chain. It uses `_immortal|v2_XDAI__{{key}}`.

## Screenshots
![image](https://user-images.githubusercontent.com/20442784/149363597-7b419461-421f-4a26-b2b7-18864e0fd290.png)
